### ISSUE TITLE
How to pass animated property to other elements

### ISSUE THREAD

**Issue by cihadturhan** [fram-x/FluidTransitions#58]
As far as I see, `animated` property is used with `shared`

What I want to do is pass the `animated` property to other components so I can customize their transition.
Is there a way to do is like:

```html
<Transition animated="progress" anchor="blaBla">
    <MyComponent />
</Transition>
```

**Comment by chrfalch** [comment]
Hi @cihadturhan! You can look at the example showing how this can be done here: 
https://github.com/fram-x/FluidTransitions/blob/develop/Examples/src/AnimatedProperty.js

**Comment by cihadturhan** [comment]
Hi @chrfalch, 
Thanks for the reply! That is different from what I wanted. Let's say you have the following scenario

```
List Screen          -> Detail Screen

Image (shared=image) -> Image (shared=image)
                     -> Custom Component (anchor=image)
```

I want my other custom component use the same animated prop with `Image`. This is not possible with the current implementation. However, I examined the code and added a small patch to do what I want:

```diff
   const element = React.Children.only(anchor.reactElement.props.children);
   const key = "an-" + anchor.name + anchor.route;
   const props = { ...element.props, __index: to.index };
+
+  if(anchor.animated) {
+    props[anchor.animated] = getInterpolationFunction(false)
+  }
+
   const component = React.createElement(element.type, { ...props, key });
-  const retVal = createAnimatedWrapper({component, nativeStyles});
+  const retVal = createAnimatedWrapper({component, nativeStyles: (anchor.animated ? [positionStyle] : nativeStyles)});
   return retVal;
 }
```

So now I can use it like:
```jsx
<Transition animated="progress" anchor="image">
    <MyComponent />
</Transition>
```

**Comment by chrfalch** [comment]
Thanks for the suggestion! I'm still a little bit unsure if I completely understand what you are trying to accomplish - maybe its just me being stupid ;) - what should happen with `MyComponent` during the transition? It should be anchored AND use the `animated` property to do some custom animation?

A PR would be great - always happy for contributions!

**Comment by cihadturhan** [comment]
> It should be anchored AND use the animated property to do some custom animation?

Exactly! By default an ‘anchor’ means the same transition with the ‘shared’ element. But let’s say I want image to reposition and my component to rotate & scale synchronously.
